cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/graph/types/HasType.kt

```diff
@@ -25,9 +25,11 @@
  */
 package de.fraunhofer.aisec.cpg.graph
 
+import de.fraunhofer.aisec.cpg.frontends.TranslationException
 import de.fraunhofer.aisec.cpg.graph.types.Type
+import java.util.Optional
 
-interface HasType {
+interface HasType : ContextProvider {
```

Comment:
I kinda don't like  having every node that has a type, hold a reference to the context, don't we just need it in the handlers? I can see that we will have the need for multiple translation contexts associated to different nodes in one result.

Reply:
We need a reference to the context, more specifically to the type manager in each node which has a type because the we are using functions of the `TypeManager` in `setType` and so on. Previously we were using the type manager singleton, not we need a reference to the appropriate type manager in each node.